compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -323,14 +323,8 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
     }
 
     private def genJSClassDispatcher(classSym: Symbol, name: JSName): js.Tree = {
-      var alts: List[Symbol] = Nil
-      for {
-        sym <- classSym.info.members
-        if !sym.isBridge && jsNameOf(sym) == name
-      } {
-        val tpe = sym.tpe
-        if (!alts.exists(alt => tpe.matches(alt.tpe)))
-          alts ::= sym
+      val alts = classSym.info.members.toList.filter { sym =>
+        !sym.isBridge && jsNameOf(sym) == name
```

Comment:
Only if the lambda is a one-liner, from its `param =>` up to the end of its body.

So this is forbidden:
```scala
foo { a => a + 1}
```
but this is OK:
```scala
foo { a =>
  a + 1
}
```